[Bug]: linode_instance creation crashes on empty authorized_keys

### Terraform Version

Terraform v1.1.7

### Linode Provider Version

v1.29.4

### Effected Terraform Resources

_No response_

### Terraform Config Files

```
resource "tls_private_key" "website" {
  algorithm   = "ECDSA"
  ecdsa_curve = "P224"  # because of this, public_key_openssh attribute is not populated => empty
}

resource "local_sensitive_file" "private_key" {
  content         = tls_private_key.website.private_key_pem
  filename        = "xxx-website-root.pem"
  file_permission = "0600"
}

data "linode_profile" "me" {}

resource "linode_instance" "website" {
  label            = "xxx-website"
  region           = "eu-central"
  type             = "g6-standard-1"
  image            = "linode/ubuntu22.04"
  authorized_keys  = [chomp(tls_private_key.website.public_key_openssh)]
  authorized_users = [data.linode_profile.me.username]
  backups_enabled  = true
  watchdog_enabled = true
}
```

### Debug Output

```
$ TF_LINODE=DEBUG LINODE_DEBUG=1 tf apply

...

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # linode_instance.website will be created
  + resource "linode_instance" "website" {
      + authorized_keys    = [
          + "",  # <<<<<<<<<<<<<<<<<<<<<<<
        ]
      + authorized_users   = [
          + "xxx",
        ]
      + backups            = (known after apply)
      + backups_enabled    = true
      + boot_config_label  = (known after apply)
      + booted             = (known after apply)
      + id                 = (known after apply)
      + image              = "linode/ubuntu22.04"
      + ip_address         = (known after apply)
      + ipv4               = (known after apply)
      + ipv6               = (known after apply)
      + label              = "xxx-website"
      + private_ip_address = (known after apply)
      + region             = "eu-central"
      + resize_disk        = false
      + shared_ipv4        = (known after apply)
      + specs              = (known after apply)
      + status             = (known after apply)
      + swap_size          = (known after apply)
      + type               = "g6-standard-1"
      + watchdog_enabled   = true

      + alerts {
          + cpu            = (known after apply)
          + io             = (known after apply)
          + network_in     = (known after apply)
          + network_out    = (known after apply)
          + transfer_quota = (known after apply)
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + website_ip = (known after apply)

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

linode_instance.website: Creating...
╷
│ Error: Plugin did not respond
│ 
│   with linode_instance.website,
│   on dev-machine.tf line 132, in resource "linode_instance" "website":
│  132: resource "linode_instance" "website" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-linode_v1.29.4 plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 25 [running]:
github.com/linode/terraform-provider-linode/linode/instance.createResource({0x2c32b88, 0xc000ba0960}, 0x0?, {0x24ab220?, 0xc00019a2c0})
        github.com/linode/terraform-provider-linode/linode/instance/resource.go:190 +0x259d
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc000371420, {0x2c32bc0, 0xc000975230}, 0xd?, {0x24ab220, 0xc00019a2c0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.23.0/helper/schema/resource.go:707 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000371420, {0x2c32bc0, 0xc000975230}, 0xc000cb0c30, 0xc000d03080, {0x24ab220, 0xc00019a2c0})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.23.0/helper/schema/resource.go:837 +0xa7a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00000df80, {0x2c32bc0?, 0xc000975110?}, 0xc000972500)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.23.0/helper/schema/grpc_provider.go:1021 +0xe3c
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc00056e820, {0x2c32bc0?, 0xc000974900?}, 0xc000c98150)
        github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/tf5server/server.go:818 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x276ad80?, 0xc00056e820}, {0x2c32bc0, 0xc000974900}, 0xc000c980e0, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000001e0, {0x2c36940, 0xc000602d00}, 0xc000576120, 0xc0005af7a0, 0x3959fc0, 0x0)
        google.golang.org/grpc@v1.48.0/server.go:1295 +0xb0b
google.golang.org/grpc.(*Server).handleStream(0xc0000001e0, {0x2c36940, 0xc000602d00}, 0xc000576120, 0x0)
        google.golang.org/grpc@v1.48.0/server.go:1636 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.48.0/server.go:932 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.48.0/server.go:930 +0x28a

Error: The terraform-provider-linode_v1.29.4 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behavior

The provider should crash gracefully saying something along the lines: "authorized_keys cannot be empty string ..."

### Actual Behavior

The provider crashes with stacktrace.

### Steps to Reproduce

Try to create linode_instance with a single authorized_key, which is empty string. (See the abovementioned TF snippet)

--- a/linode/instance/helpers.go
+++ b/linode/instance/helpers.go
@@ -407,12 +407,22 @@ func createInstanceDisk(
 
 		if authorizedKeys, ok := disk["authorized_keys"]; ok {
 			for _, sshKey := range authorizedKeys.([]interface{}) {
+				if sshKey == nil {
+					return nil, fmt.Errorf(
+						"invalid input for disk authorized_keys: keys cannot be empty or null")
+				}
+
 				diskOpts.AuthorizedKeys = append(diskOpts.AuthorizedKeys, sshKey.(string))
 			}
 		}
 
 		if authorizedUsers, ok := disk["authorized_users"]; ok {
 			for _, user := range authorizedUsers.([]interface{}) {
+				if user == nil {
+					return nil, fmt.Errorf(
+						"invalid input for disk authorized_users: users cannot be empty or null")
+				}
+
 				diskOpts.AuthorizedUsers = append(diskOpts.AuthorizedUsers, user.(string))
 			}
 		}

--- a/linode/instance/resource.go
+++ b/linode/instance/resource.go
@@ -187,10 +187,18 @@ func createResource(ctx context.Context, d *schema.ResourceData, meta interface{
 	// If we don't have disks and we don't have configs, use the single API call approach
 	if !disksOk && !configsOk {
 		for _, key := range d.Get("authorized_keys").([]interface{}) {
+			if key == nil {
+				return diag.Errorf("invalid input for authorized_keys: keys cannot be empty or null")
+			}
+
 			createOpts.AuthorizedKeys = append(createOpts.AuthorizedKeys, key.(string))
 		}
-		for _, key := range d.Get("authorized_users").([]interface{}) {
-			createOpts.AuthorizedUsers = append(createOpts.AuthorizedUsers, key.(string))
+		for _, user := range d.Get("authorized_users").([]interface{}) {
+			if user == nil {
+				return diag.Errorf("invalid input for authorized_users: users cannot be empty or null")
+			}
+
+			createOpts.AuthorizedUsers = append(createOpts.AuthorizedUsers, user.(string))
 		}
 		createOpts.RootPass = d.Get("root_pass").(string)
 		if createOpts.RootPass == "" {

--- a/linode/instance/resource_test.go
+++ b/linode/instance/resource_test.go
@@ -155,6 +155,30 @@ func TestAccResourceInstance_authorizedUsers(t *testing.T) {
 	})
 }
 
+func TestAccResourceInstance_validateAuthorizedKeys(t *testing.T) {
+	t.Parallel()
+
+	instanceName := acctest.RandomWithPrefix("tf_test")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.TestAccProviders,
+		CheckDestroy: acceptance.CheckInstanceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: tmpl.AuthorizedKeysEmpty(t, instanceName, testRegion),
+				ExpectError: regexp.MustCompile(
+					"invalid input for authorized_keys"),
+			},
+			{
+				Config: tmpl.DiskAuthorizedKeysEmpty(t, instanceName, testRegion),
+				ExpectError: regexp.MustCompile(
+					"invalid input for disk authorized_keys"),
+			},
+		},
+	})
+}
+
 func TestAccResourceInstance_interfaces(t *testing.T) {
 	t.Parallel()
 

--- a/linode/instance/tmpl/template.go
+++ b/linode/instance/tmpl/template.go
@@ -376,6 +376,24 @@ func AuthorizedUsers(t *testing.T, label, pubKey, region string) string {
 		})
 }
 
+func AuthorizedKeysEmpty(t *testing.T, label, region string) string {
+	return acceptance.ExecuteTemplate(t,
+		"instance_authorized_keys_empty", TemplateData{
+			Label:  label,
+			Image:  acceptance.TestImageLatest,
+			Region: region,
+		})
+}
+
+func DiskAuthorizedKeysEmpty(t *testing.T, label, region string) string {
+	return acceptance.ExecuteTemplate(t,
+		"instance_disk_authorized_keys_empty", TemplateData{
+			Label:  label,
+			Image:  acceptance.TestImageLatest,
+			Region: region,
+		})
+}
+
 func StackScript(t *testing.T, label, region string) string {
 	return acceptance.ExecuteTemplate(t,
 		"instance_stackscript", TemplateData{

--- a/linode/instance/tmpl/templates/authorized_keys_empty.gotf
+++ b/linode/instance/tmpl/templates/authorized_keys_empty.gotf
@@ -1,0 +1,14 @@
+{{ define "instance_authorized_keys_empty" }}
+
+resource "linode_instance" "foobar" {
+    label = "{{.Label}}"
+    group = "tf_test"
+    type = "g6-nanode-1"
+    image = "{{.Image}}"
+    region = "{{ .Region }}"
+    authorized_keys = [
+        ""
+    ]
+}
+
+{{ end }}

--- a/linode/instance/tmpl/templates/disk_authorized_keys_empty.gotf
+++ b/linode/instance/tmpl/templates/disk_authorized_keys_empty.gotf
@@ -1,0 +1,17 @@
+{{ define "instance_disk_authorized_keys_empty" }}
+
+resource "linode_instance" "foobar" {
+    label = "{{.Label}}"
+    type = "g6-nanode-1"
+    region = "{{ .Region }}"
+
+    disk {
+        label = "boot"
+        size = 4096
+        image = "{{.Image}}"
+        authorized_keys = [""]
+        root_pass = "myc00lp@ss!"
+    }
+}
+
+{{ end }}

